llvm/lib/IR/RepoTicket.cpp

```diff
@@ -116,7 +116,39 @@ static GODigestState accumulateGOFinalDigestOrInitialDigestAndGetGODigestState(
   return accumulateGOInitialDigestAndGetGODigestState(GO, GOHash, GOIMap);
 }
 
-// Update the GO's hash value by adding the hash of its dependents.
+static bool updateDigestUseContributions(const GlobalObject *GO, MD5 &GOHash,
+                                         const GOVec &Contributions,
+                                         GOInfoMap &GOIMap) {
+  for (const GlobalObject *const G : Contributions) {
+    assert(isa<GlobalVariable>(GO) &&
+           "Only global variables can have contributions!");
+    assert(isa<llvm::Function>(G) && "All contributions are functions!");
+    GOHash.update(GOIMap[G].InitialDigest.Bytes);
+  }
+  return !Contributions.empty();
+}
+
+template <typename Function>
+static bool updateDigestUseDependencies(MD5 &GOHash, const GOVec &Dependencies,
+                                        GOStateMap &Visited, GOInfoMap &GOIMap,
+                                        Function AccumulateGODigest) {
+  bool Changed;
```

Comment:
I think that Changed should be initialised in case Dependencies is empty.

Reply:
Yes, `Changed` should be initialised. However, it is not decided by whether the Dependencies is empty or not. Dependencies is different from Contributions. Contributions must be defined functions (not declaration). Therefore, the GO’s hash will be changed if its contributions is not empty. However, Dependencies could include the declared functions (e.g. printf), which don’t have the digest and will not change the GO’s hash. Therefore, `Changed` is initialised to `false`.